### ISSUE TITLE
2FA authentication question

### ISSUE THREAD

**Issue by stephanoshadjipetrou** [agentejo/cockpit#1435]
Hello! Just a question: is there any way to add/enable any 2FA for login?

**Comment by laphilipa** [comment]
There is nothing bild into cockpit directly yet. Here are some official plugins https://discourse.getcockpit.com/t/list-of-cockpit-addons/234 among them Auth0 I belive you could build 2FA with them. I wouldn't recommend giving controll of authentication to a service provider though.

**Comment by laphilipa** [comment]
@stephanoshadjipetrou have you seen this plugin https://github.com/agentejo/2FA?

**Comment by stephanoshadjipetrou** [comment]
@laphilipa thanks! how could I use this in cockpit?

**Comment by laphilipa** [comment]
Just clone the git repo in the addons direcotry of your cockpit installation. Cockpit will then detect it automatically. In your account settings on the bottom you get a QR code for the 2FA app.

**Comment by stephanoshadjipetrou** [comment]
thanks @laphilipa! cloned the repo in addons dir as you said, but when navigating to my account page get a 500 error

**Comment by laphilipa** [comment]
Are you using the official docker image or are you deploying some other way? Is there an error message with more information? Usually cockpit displays the error on the same site it happened if not try checking the apache/ngix logs.

**Comment by stephanoshadjipetrou** [comment]
Nope. Following the "Traditional Installation" from [here](https://getcockpit.com/documentation/getting-started/installation). Also not much on the error returned, just the document.title renders "Internal Server Error". But ye, will check the nginx logs to see if can find anything related. Thanks once again :)

**Comment by patrickuhlmann** [comment]
I tested it and it worked for me. I wrote instructions that you find here: https://github.com/agentejo/2FA/pull/1/files

**Comment by stephanoshadjipetrou** [comment]
@patrickuhlmann nice! yes that's handy :)

but still not working for me. followed the instructions and this what see in /accounts/account page
![error](https://user-images.githubusercontent.com/11508354/115673543-6e1eec00-a355-11eb-8524-1110e266c8f3.png)

checked in console, nginx access & error logs, php logs and couldn't find something related to this.

**Comment by siemvaessen** [comment]
@stephanoshadjipetrou any update on this?
